cmd/agent/main.go

```diff
@@ -42,14 +42,14 @@ func main() {
 		os.Exit(0)
 	}
 
-	logFilePath, err := filepath.Abs("cirrus-agent.log")
-	if err != nil {
-		log.Printf("Failed to get absolute log file path: %v", err)
-		logFilePath = "cirrus-agent.log"
-	}
-	logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0660)
+	logFilePath := filepath.Join(os.TempDir(), "cirrus-agent.log")
+	defer uploadAgentLogs(logFilePath, *taskIdPtr, *clientTokenPtr)
+
+	logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0660)
```

Comment:
Why not use https://golang.org/pkg/io/ioutil/#TempFile instead?

Reply:
To have a predictable name so if agent terminates unexpectedly it's always clear where to find the logs.

Reply:
Unfortunately this wouldn't play nice with multiple agents running on the same host (e.g. spawned by two separate persistent workers).

But isn't uploading the logs the cloud already solves the log retrieval problem? Otherwise a new command-line argument to specify a separate log path should probably be added.

Reply:
Included task id in the file name which is removed upon a successful upload.